src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -495,6 +537,11 @@ public AllocatedEndpoint(EndpointAnnotation endpoint, string address, int port,
         public override string ToString() { throw null; }
     }
 
+    public static partial class AspireStoreExtensions
+    {
+        public static IAspireStore CreateStore(this IDistributedApplicationBuilder builder) { throw null; }
```

Comment:
This API means you can't create the aspire store in a lifecycle hook (where there's no access to the builder). 

cc @sebastienros

Reply:
Do we have concrete scenarios where we know you'd need it before build is called? I thought the same while working on the dev cert stuff before I realized all the real code was in event lifecycle callbacks anyway so I'd have access to services.

Reply:
@eerhardt is this resolved?

Reply:
It is resolved now with https://github.com/dotnet/aspire/pull/7693